src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -86,7 +88,8 @@ public static void Initialize()
             if (TracerSettings.LoadTracerAtStartup || MeterSettings.LoadMetricsAtStartup)
             {
                 // Initialize SdkSelfDiagnosticsEventListener to create an EventListener for the OpenTelemetry SDK
-                _sdkEventListener = new(EventLevel.Warning);
+                _sdkEventListener = new(EventLevel.Warning, Logger);
+                _lazyInstrumentation = new(new LazyInstrumentationBuilders(TracerSettings));
```

Comment:
How lazily does it load? I could see the library load before the AddSource being called.

Reply:
Unfortunately I didn't have better naming for this - could be bad choice. 
I'm afraid that ASP.NET apps at least will load the framework dlls quite early in the bootstrapping process. This could not be the same behaviour with other instrumentations - specially the ones that are dynamically loaded.